ml-agents/mlagents/trainers/settings.py

```diff
@@ -199,28 +199,37 @@ def structure(d: Mapping, t: type) -> Any:
             enum_key = RewardSignalType(key)
             t = enum_key.to_settings()
             d_final[enum_key] = strict_to_cls(val, t)
+            if "encoding_size" in val:
+                logger.warning(
+                    "'encoding_size' was deprecated for RewardSignals. Please use network_settings."
+                )
+                # If network settings was not specified, use the encoding size. Otherwise, use hidden_units
+                if "network_settings" not in val:
+                    d_final[enum_key].network_settings.hidden_units = val[
+                        "encoding_size"
+                    ]
         return d_final
 
 
 @attr.s(auto_attribs=True)
 class GAILSettings(RewardSignalSettings):
-    encoding_size: int = 64
     learning_rate: float = 3e-4
+    encoding_size: int = 0
```

Comment:
need to keep so cattrs doesnt error if its specified.

Reply:
Will it break if it's Optional[int] = None?

Reply:
Negative. Updated.